sdk/src/main/java/com/horizen/box/WithdrawalRequestBox.java

```diff
@@ -59,36 +34,13 @@ public BoxSerializer serializer() {
 
     @Override
     public byte[] bytes() {
-        return Bytes.concat(proposition.bytes(), Longs.toByteArray(nonce), Longs.toByteArray(value));
+        return Bytes.concat(Longs.toByteArray(nonce), WithdrawalRequestBoxDataSerializer.getSerializer().toBytes(boxData));
     }
 
     public static WithdrawalRequestBox parseBytes(byte[] bytes) {
-        MCPublicKeyHashProposition t = MCPublicKeyHashProposition.parseBytes(Arrays.copyOf(bytes, MCPublicKeyHashProposition.getLength()));
-        long nonce = Longs.fromByteArray(Arrays.copyOfRange(bytes, MCPublicKeyHashProposition.getLength(), MCPublicKeyHashProposition.getLength() + 8));
-        long value = Longs.fromByteArray(Arrays.copyOfRange(bytes, MCPublicKeyHashProposition.getLength() + 8, MCPublicKeyHashProposition.getLength() + 16));
-        return new WithdrawalRequestBox(t, nonce, value);
-    }
-
-    @Override
-    public int hashCode() {
-        return Arrays.hashCode(this.proposition.bytes());
-    }
+        long nonce = Longs.fromByteArray(Arrays.copyOf(bytes, Longs.BYTES));
```

Comment:
I still prefer to do it like a "constructor", i.e. all work with a nonce (including serialization/deserialization) do in AbstractNoncedBox